test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
     end
 end
 
-# transpose of SubArrays
-A = view(sprandn(10, 10, 0.3), 1:4, 1:4)
-@test  transpose(Array(A)) == Array(transpose(A))
-@test ctranspose(Array(A)) == Array(ctranspose(A))
-
-# exp
-A = sprandn(5,5,0.2)
-@test e.^A ≈ e.^Array(A)
-
-# reductions
-pA = sparse(rand(3, 7))
+@testset "Transpose of subarrays" begin
+    A = view(sprandn(10, 10, 0.3), 1:4, 1:4)
+    @test  transpose(Array(A)) == Array(transpose(A))
+    @test ctranspose(Array(A)) == Array(ctranspose(A))
+end
 
-for arr in (se33, sA, pA)
-    for f in (sum, prod, minimum, maximum, var)
-        farr = Array(arr)
-        @test f(arr) ≈ f(farr)
-        @test f(arr, 1) ≈ f(farr, 1)
-        @test f(arr, 2) ≈ f(farr, 2)
-        @test f(arr, (1, 2)) ≈ [f(farr)]
-        @test isequal(f(arr, 3), f(farr, 3))
-    end
+@testset "Exp" begin
+    A = sprandn(5,5,0.2)
+    @test e.^A ≈ e.^Array(A)
 end
 
-for f in (sum, prod, minimum, maximum)
-    # Test with a map function that maps to non-zero
+@testset "Reductions" begin
+    pA = sparse(rand(3, 7))
+
     for arr in (se33, sA, pA)
-        @test f(x->x+1, arr) ≈ f(arr+1)
+        for f in (sum, prod, minimum, maximum, var)
+            farr = Array(arr)
+            @test f(arr) ≈ f(farr)
+            @test f(arr, 1) ≈ f(farr, 1)
+            @test f(arr, 2) ≈ f(farr, 2)
+            @test f(arr, (1, 2)) ≈ [f(farr)]
+            @test isequal(f(arr, 3), f(farr, 3))
+        end
     end
 
-    # case where f(0) would throw
-    @test f(x->sqrt(x-1), pA+1) ≈ f(sqrt.(pA))
-    # these actually throw due to #10533
-    # @test f(x->sqrt(x-1), pA+1, 1) ≈ f(sqrt(pA), 1)
-    # @test f(x->sqrt(x-1), pA+1, 2) ≈ f(sqrt(pA), 2)
-    # @test f(x->sqrt(x-1), pA+1, 3) ≈ f(pA)
-end
-
-# empty cases
-@test sum(sparse(Int[])) === 0
-@test prod(sparse(Int[])) === 1
-@test_throws ArgumentError minimum(sparse(Int[]))
-@test_throws ArgumentError maximum(sparse(Int[]))
-@test var(sparse(Int[])) === NaN
+    for f in (sum, prod, minimum, maximum)
+        # Test with a map function that maps to non-zero
+        for arr in (se33, sA, pA)
+            @test f(x->x+1, arr) ≈ f(arr+1)
+        end
 
-for f in (sum, prod, minimum, maximum, var)
-    @test isequal(f(spzeros(0, 1), 1), f(Array{Int}(0, 1), 1))
-    @test isequal(f(spzeros(0, 1), 2), f(Array{Int}(0, 1), 2))
-    @test isequal(f(spzeros(0, 1), (1, 2)), f(Array{Int}(0, 1), (1, 2)))
-    @test isequal(f(spzeros(0, 1), 3), f(Array{Int}(0, 1), 3))
+        # case where f(0) would throw
+        @test f(x->sqrt(x-1), pA+1) ≈ f(sqrt.(pA))
+        # these actually throw due to #10533
+        # @test f(x->sqrt(x-1), pA+1, 1) ≈ f(sqrt(pA), 1)
+        # @test f(x->sqrt(x-1), pA+1, 2) ≈ f(sqrt(pA), 2)
+        # @test f(x->sqrt(x-1), pA+1, 3) ≈ f(pA)
+    end
 end
 
-# spdiagm
-@test Array(spdiagm((ones(2), ones(2)), (0, -1), 3, 3)) ==
-                       [1.0  0.0  0.0; 1.0  1.0  0.0;  0.0  1.0  0.0]
-@test Array(spdiagm(ones(2), -1, 3, 3)) == diagm(ones(2), -1)
+@testset "Empty cases" begin
```

Comment:
These tests are a subset of the reduction tests above. Please restore them to their original location.